HOMEWORK.js

```diff
@@ -58,31 +58,36 @@ function homework (renderList) {
 	// ◽ super bonus: remove the X if it already has one (put it back to incomplete)
 	function onCompleteTodo (index) {
 		// Add code here
-
+		todos[index].charAt(0) === 'X' ? todos[index] = todos[index].slice(1) : todos[index] = 'X' + todos[index];
 		renderList(todos);
 	}
 
 	// ✅This function should:
 	// ◽ remove from the list all of the completed items (that start with 'X')
 	function onRemoveCompleted () {
 		// Add code here
-
+		todos.filter(item => item.charAt(0) === 'X').map(item => todos.splice(todos.indexOf(item),1));
 		renderList(todos);
 	}
 
 	// ✅This function should:
 	// ◽ add an exclamation mark ('!') to the end of every item in the list
 	function onUrgent () {
 		// Add code here
-		
+		todos.forEach(item => todos[todos.indexOf(item)] = item + "!");
```

Comment:
This works, but what you have done here is reinvented `map`!

Don't forget, you can set the todos with `=` like you did in your `onClearAll` function.